components/script/dom/globalscope.rs

```diff
@@ -3167,16 +3173,19 @@ impl GlobalScope {
                 task!(gamepad_disconnected: move || {
                     let global = this.root();
                     if let Some(window) = global.downcast::<Window>() {
-                        let gamepad_list = window.Navigator().gamepads();
-                        if let Some(gamepad) = gamepad_list.Item(index as u32) {
-                            if window.Document().is_fully_active() {
-                                gamepad.update_connected(false, gamepad.exposed());
-                                gamepad_list.remove_gamepad(index);
+                        let navigator = window.Navigator();
+                        let mut gamepad_list = navigator.gamepads();
+                        if let Some(gamepad_or_none) = gamepad_list.get(index) {
+                            if let Some(gamepad) = gamepad_or_none {
+                                if window.Document().is_fully_active() {
+                                    gamepad.update_connected(false, gamepad.exposed());
+                                    gamepad_list.remove(index);
+                                }
                             }
                         }
-                        for i in (0..gamepad_list.Length()).rev() {
-                            if gamepad_list.Item(i as u32).is_none() {
-                                gamepad_list.remove_gamepad(i as usize);
+                        for i in (0..gamepad_list.len()).rev() {
+                            if gamepad_list.get(i as usize).is_none() {
+                                gamepad_list.remove(i as usize);
```

Comment:
instead of removing it, it should be set to `None`.

Reply:
This is step 6 of the queued task though, "While navigator.[[gamepads]] is not empty and the last item of navigator.[[gamepads]] is null, remove the last item of navigator.[[gamepads]]."

Reply:
Ah, ok!